pkg/api/types.go

```diff
@@ -2699,8 +2699,9 @@ type FinalizerName string
 
 // These are internal finalizer values to Kubernetes, must be qualified name unless defined here
 const (
-	FinalizerKubernetes FinalizerName = "kubernetes"
-	FinalizerOrphan     string        = "orphan"
+	FinalizerKubernetes       FinalizerName = "kubernetes"
+	FinalizerOrphanDependents string        = "orphan"
+	FinalizerDeleteDependents string        = "DeletingDependents"
```

Comment:
And "kubernetes" is definitely not a verb (yet). (also that makes little sense as a finalizer name but that's an issue for another PR.)